coderd/database/databasefake/databasefake.go

```diff
@@ -1717,6 +1717,46 @@ func (q *fakeQuerier) GetTemplateVersionByJobID(_ context.Context, jobID uuid.UU
 	return database.TemplateVersion{}, sql.ErrNoRows
 }
 
+func (q *fakeQuerier) GetPreviousTemplateVersion(_ context.Context, arg database.GetPreviousTemplateVersionParams) (database.TemplateVersion, error) {
+	q.mutex.RLock()
+	defer q.mutex.RUnlock()
+
+	var currentTemplateVersion database.TemplateVersion
+	for _, templateVersion := range q.templateVersions {
+		if templateVersion.TemplateID != arg.TemplateID {
+			continue
+		}
+		if templateVersion.Name != arg.Name {
+			continue
+		}
+		if templateVersion.OrganizationID != arg.OrganizationID {
+			continue
+		}
+		currentTemplateVersion = templateVersion
+		break
+	}
+
+	previousTemplateVersions := make([]database.TemplateVersion, 0)
+	for _, templateVersion := range q.templateVersions {
+		if templateVersion.ID == currentTemplateVersion.ID {
```

Comment:
I guess we should do this too, to ensure correctness?

```suggestion
		if templateVersion.ID == currentTemplateVersion.ID || templateVersion.OrganizationID != arg.OrganizationID || templateVersion.TemplateID != arg.TemplateID {
```